djAerolith/wordwalls/templates/wordwalls/stats.html

```diff
@@ -0,0 +1,253 @@
+{% extends "base.html" %}
+{% load i18n %}
+{% block ss %}
+    <link href="/static/css/wordwalls/wordwallsCreateTable.css" rel="stylesheet">
+    <link href="/static/css/aerolith/bootstrap-datepicker.css" rel="stylesheet">
+{% endblock %}
+{% block title %}{% trans "Wordwalls - select list" %}{% endblock %}
+
+{% block jquery_js %}{% endblock %}
+
+{% block content %}
+<script src="/static/js/aerolith/jquery-1.11.0.min.js"></script>
+<script src="/static/lib/bootstrap/js/bootstrap.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.1.4/Chart.bundle.min.js"></script>
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
+
+
+<div id="div1">
+
+    <canvas id="scoresChart" width="400" height="100"></canvas>
+
+</div>
+
+<p>
+
+<div id="div2">
+
+    <canvas id="timeChart" width="400" height="100"></canvas>
+
+</div>
+
+</p>
+
+<p>
+
+Lexicon:<br>
+<select class="form-control" id="lexicon">
+  <option value="1">CSW15</option>
+  <option value="4">OWL2</option>
+  <option value="6">CSW12</option>
+  <option value="7">America</option>
+  <option value="8">FISE09</option>
+</select>
+
+<p>
+
+Challenges:<br>
+<select class="form-control" id="challenge">
+  <option value="6">Today's 7's</option>
+  <option value="7">Today's 8's</option>
+</select>
+
+</p>
+
+<button type="button" class="btn btn-primary" id="dropdown-button">Submit</button>
+
+</p>
+
+<script>
+
+// These hold info for the charts
+var scoresByDate = {};
+var dateArray = [];
+var timeRemainingByDate = {};
+
+$( "#dropdown-button" ).click(function() {
+    $("#scoresChart").load();
+
+    // This is the info we want
+    var lexicon = $('#lexicon').val()
+    var challenge = $('#challenge').val()
+
+    // Call the API, receive JSON
+    $.ajax({
+          method: "GET",
+          url: "/wordwalls/stats/api/" + lexicon + "/" + challenge,
+          dataType: 'json'
+    })
+        .done(function(msg) {
+
+            // Parse the JSON for the info we want for the charts
+            for (i = 0; i < msg.length; i++) {
+                var x = i;
+                var value = msg[i];
+                var dateString = value.Date;
+                var dateYearOnly = dateString.slice(0, 4);
+                var scoreShortened = (Number(value.Score) / Number(value.maxScore)) * 100;
+
+                scoresByDate[value.Date] = scoreShortened.toFixed(2);
+                timeRemainingByDate[value.Date] = value.timeRemaining;
+
+                // dateArray.push(dateYearOnly);
+                dateArray.push(value.Date);
+
+            }
+
+            // Call the charts function to generate the charts on the page
+            charts();
+        });
+
+function isDateInArray(value, array) {
+    return array.indexOf(value) > -1;
+}
+
+function charts () {
+
+    // Info for chart axes
+    dateArray = dateArray.sort();
```

Comment:
`dateArray.sort()` does not need to be assigned back to `dateArray` since it sorts in place